scrapy_zyte_api/providers.py

```diff
@@ -120,6 +120,10 @@ async def __call__(  # noqa: C901
                         "e.g. Annotated[Actions, actions([...list of actions...])]."
                     )
                 zyte_api_meta["actions"] = [dict(action) for action in cls.__metadata__[0]]  # type: ignore[attr-defined]
```

Comment:
This is the line that makes the reverse conversion, it needs to be updated to be recursive too.